### PR TITLE
Added a --config flag

### LINKED ISSUE #242: Config File path flag
Being able to point at a config flag (`--config` for example) in addition to the path search locations that are in place would be helpful.

Would simplify using hadolint in purpose-built containers

### PR DESCRIPTION
**Pull request by lorenzo**
This allows users to pass a custom confuguration file path.

closes #242

### CODE REVIEW

File: app/Main.hs [hunk 1]
```diff
@@ -80,13 +81,21 @@ parseOptions :: Parser LintOptions
...
     parseRulesConfig
   where
     version = switch (long "version" <> short 'v' <> help "Show version")
     --
+    -- | Parse the config filename to use
+    configFile =
+        optional
+            (strOption
+                 (long "config" <> short 'c' <> metavar "FILENAME" <>
+                  help "Use a custom file as configuration"))
```
Comment (zemanlx) [inline app/Main.hs]: Would you mind to change it to `Path to the configuration file`?

File: README.md [hunk 2]
```diff
@@ -107,6 +107,13 @@ file, in order or preference:
 
 In windows, the `%LOCALAPPDATA%` environment variable is used instead of `XDG_CONFIG_HOME`
 
+Additionally, you can pass a custom configuration file in the command line with
+the `--config` option
+
+```bash
+hadolint --config /file/to/config.yaml Dockerfile
```
[the remark this answers is not in the document]
Comment (lorenzo) [reply README.md]: Intentional, you need to write up to the file name. The use case is that you may have different config files in the same folder.

File: app/Main.hs (hunk 1 above)
Comment (lorenzo) [reply app/Main.hs]: It is not just the path to, it needs to include the file name

File: README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (lorenzo) [reply README.md]: ah, I' m dumb... I understand what you mean now, will change

File: app/Main.hs (hunk 1 above)
Comment (zemanlx) [reply app/Main.hs]: fair enough

**Comment by lorenzo** [comment]
> I think that the is correct because we specify exactly one in CLI.

We can specify many docker registries in the cli, like we can specify many ignore rules

**Comment by zemanlx** [comment]
@lorenzo yes, but only one per option
```bash
$ hadolint --trusted-registry lol.com phew.org docker/Dockerfile
hadolint: phew.org: openBinaryFile: does not exist (No such file or directory)

$ hadolint --trusted-registry lol.com --trusted-registry phew.org docker/Dockerfile
docker/Dockerfile:1 DL3026 Use only an allowed registry in the FROM image
docker/Dockerfile:31 DL3026 Use only an allowed registry in the FROM image
```

**Comment by lorenzo** [comment]
@zemanlx I changed some help texts, is it better now?

**Pull request merged**